designs/2023-only-run-reporting-rules/README.md

```diff
@@ -0,0 +1,137 @@
+from 4 minutes and 39 seconds to just 22 seconds. As the `warn` rules are entirely ignored
+in this situation, this is a substantial amount of wasted time and resources that could be
+avoided.
+
+## Detailed Design
+
+The proposed implementation is to modify the `--quiet` flag so that ESLint will skip
+running any rules that are set to the `warn` level.
+
+From an API perspective, this would be implemented by a predicate function that acts as
+a filter on which rules should be run. The predicate would take the rule configuration,
+and return whether the rule should be run. For simplicity, this function should not be
+given rules marked as `off`, as if this function handled existing behavior then users of the
+API would have to mimic that when attempting to extend it.
+
+In `cli.js`'s `translateOptions` function, the `rulePredicate` option should be assigned to
+a predicate that checks for a `severity` of 2 (error) when the `--quiet` flag is applied, otherwise
+always returns true. In `eslint/flat-eslint.js`, the `rulePredicate` should be taken from
+the `eslintOptions` object, and passed down to the `linter.verifyAndFix` call.
+
+Within `linter.js`, the API should be added to `VerifyOptions`, and will be passed down into and
+utilized within the `runRules` function during the `configuredRules` loop, after the check
+for disabled rules and the rule existing. The `ruleId`, `rule`, rule configuration from `getRuleOptions`,
+and `severity` should be passed into the predicate function as an object. `normalizeVerifyOptions`
+should verify that the `rulePredicate` option is a function, and replace it with an always-true
+function if not. `processOptions` in `eslint-helpers.js` should also perform a validation check
+that the `rulePredicate` option is a function.
+
+The alteration to the `--quiet` flag would be implemented by passing a predicate function
+to the API that only returns true for rules that have been set to `error`, thus filtering
+out any that are marked as `warn`.
+
+The check for unused `eslint-disable` directives (`--report-unused-disable-directives`)
+should continue to mark `warn` rules as used, even when running with `--quiet`. As the
+rules are not actually run, an assumption would have to be made that all directives
+on rules marked `warn` are used when in this mode. This is a reasonable assumption, as
+the user likely does not expect `warn` flags to be touched at all in this mode.
+
+In cases of conflicting flags such as the `--max-warnings` flag, this altered `--quiet` flag
+behavior should be disabled while it is in use. This is to ensure that the `--max-warnings`
+flag continues to work as expected. The predicate API should not be set, and existing
+behavior of filtering the output should be used. This altered behavior should output to the
+console on run that `--quiet` is slower when used alongside flags that require warnings to be
+run, as well as documented.
```

Comment:
I think we shouldn't output this info, because what could the users do about it? If they're using the `--max-warnings` functionality, then the rules must be run.

Reply:
I agree, lets not confuse users with extra stdout's, because for an existing user the behaviour wouldn't change, only performance would be better.